custom/icds_reports/static/js/angular-services/locations.service.js

```diff
@@ -243,14 +244,14 @@ window.angular.module('icdsApp').factory('locationsService', ['$http', '$locatio
                     } else {
                         locationsCache[item.location_id] = data.locations;
                         vm.selectedLocations[level + 1] = data.locations[0].location_id;
-                        vm.onSelect(data.locations[0], level + 1);
+                        this.onSelectLocation(data.locations[0], level + 1);
```

Comment:
this function accepts two more arguments.  Are those skipped intentionally here? I think this would give error here as those two arugments `locationsCache, vm` are used inside the function.